### ISSUE TITLE
I want ot be able to just quick install program like in a packagemanager with a single command insted of plugin add then install then add to glabla

### ISSUE THREAD

**Issue by yodatak** [asdf-vm/asdf#1616]
### Is your feature request related to a problem? Please describe

I have a lot of tool to install so i wan to be able to have a cli simple instead of 
asdf plugin add $tooltoinstall && asdf install $tooltoinstall latest && asdf global $tooltoinstall latest

and when i try asdf on first run i want to be able to just asdf install programname and by default it add or propose to add the plugin and make it local or global by default.

### Describe the proposed solution

I want to be able to have some command like 

asdf install just --global --pluginadd
or somethink like this by default 
asdf install just 
(it seak if a plugin exist localy and then if not check remote and propose to add with y/n question if no --addplugin is )
then it install by default localy or global ? ( by default its global if you don't set --local by exemple)
and it installed just the plugin and it make it globaly in the file 

or take a file in yaml format as list to install 

### Describe similar `asdf` features and why they are not sufficient

with asdf latest --all we can show the missing but no command to install them all 

when i do this 
asdf plugin add $tooltoinstall && asdf install $tooltoinstall latest && asdf global $tooltoinstall latest
sometimes it don't do the full command and the asdf global just latest don't echo anything so imposible to know what if its good or not 

### Describe other workarounds you've considered

i describe in previous comment

**Comment by JuryA** [comment]
I agree with @yodatak. Simplifying the installation process for tools in **asdf** would be a significant improvement. It's essential to have a more streamlined way to add plugins and install them globally or locally. Your proposed solution looks pretty reasonable.

The idea of introducing a command like `asdf install just --global --pluginadd` or similar is excellent. This would significantly reduce the manual steps required to set up a tool. Here's how I envision your proposal working:

1. `asdf install just` should check if the plugin for "just" exists locally. If not, it should then check the remote repository and, if found, propose to add it with a yes/no question.

2. By default, it should install the plugin locally unless `--global` is specified.

3. After successfully installing the plugin, it should make it global by default, but users should have the option to specify `--local` if they want it to be local instead.

Additionally, your suggestion of using a YAML file as a list to install multiple tools at once is a great idea. This would be a fantastic feature for users who need to set up multiple tools quickly.

I also appreciate you pointing out the limitations of the current `asdf` features, especially the lack of a command to install all missing plugins at once. It can be frustrating when commands like `asdf global` don't provide clear feedback on whether they were successful or not.

Your proposed changes would certainly make **asdf** more user-friendly and efficient. I hope the maintainers of **asdf** consider implementing these improvements. Thank you for bringing this issue to their attention!

**Comment by hyperupcall** [comment]
The use of a YAML file is a non-starter since it's almost impossible to parse in Bash (or it would needlessly require an external dependency). #829 (and #240) is a related issue, to specify a `.tool-plugins` file for easier installation of all plugins.

I generally agree with your points, the asdf team should consider fixing issues that would significantly enhance the developer experience.

**Comment by hyperupcall** [comment]
@yodatak And in the future, _please_ properly format your text. The way you combine the CLI, its output, and normal text makes it extremely difficult to read. If it were up to me, your issue would have been closed due to lack of effort (not to mention the typo and the verboseness of the title). I'm letting you know so you don't make the same mistake for other repositories
